src/background/background-message-distributor.ts

```diff
@@ -41,8 +50,11 @@ export class BackgroundMessageDistributor {
             this.logger.log('Unable to interpret message - ', message);
         }
 
-        if (response) {
-            return response;
+        if (globalContextResult || tabContextResult || backchannelMessageResponse) {
+            return this.createResponsePromise(
```

Comment:
Is it actually the case that multiple interpreters ever respond to the same message? I know the `postMessageContentHandler`-bound messages should always be mutually exclusive with interpreter-handled messages, but I'm not sure without investigation whether there are any cases where both a global and a tab-specific interpreter have a handler for the same message.

If a message is only ever bound for exactly one of the 3 interpreters, I think it simplifies this significantly

Reply:
There are a few messages that have listeners registered with both the global and tab interpreters (AssessmentScanCompleted, StartOverAllAssessments, StartOverTest, and DetailsView.select)